src/util/address.rs

```diff
@@ -81,7 +84,8 @@ impl fmt::Display for Error {
             Error::EmptyBech32Payload => write!(f, "the bech32 payload was empty"),
             Error::InvalidBech32Variant => write!(f, "invalid bech32 checksum variant"),
             Error::InvalidWitnessVersion(v) => write!(f, "invalid witness script version: {}", v),
-            Error::UnparsableWitnessVersion => f.write_str("Incorrect format of a witness version byte"),
+            Error::UnparsableWitnessVersion(ref e) => write!(f, "Incorrect format of a witness version byte: {}", e),
```

Comment:
The error types that follow `std::error::Error` should not display their sources and leave it to the callers. That way error sources can be formatted differently. E.g. `cargo` separates them by newlines as opposed to colons.

Note that this only applies to sources that themselves implement errors and return `Some(inner)`. If error carries some additional context (paths, strings...) then this context should be displayed.

Reply:
Well, everywhere throughout the library it is done this way (you can also check the lines above), so changing this method should be a topic for the separate PR

Reply:
Fixed in 6848ea43aa53e903323408bfe78bec342dc9efa4